src/hooks/useStatedBean.ts

```diff
@@ -1,39 +1,85 @@
-import { EffectContext } from '../core';
+import { EffectContext, StatedBeanScope, StatedBeanContainer } from '../core';
 import { getStatedBeanContext } from '../context';
 import { ClassType } from '../types/ClassType';
 
-import { useContext, useState, useEffect } from 'react';
+import { useContext, useState, useEffect, useCallback } from 'react';
+
+export interface UseStatedBeanOption {
+  dependentFields: Array<string | symbol>;
+  scope: StatedBeanScope;
+}
 
 export function useStatedBean<T extends ClassType>(
   type: T,
-  dependentFields?: Array<string | symbol>,
-): InstanceType<T> {
+  option: Partial<UseStatedBeanOption> = {
+    dependentFields: [],
+    scope: StatedBeanScope.DEFAULT,
+  },
+): InstanceType<T> | undefined {
```

Comment:
`useStatedBean`‘s return type may be undefined now. this is a block change.   @JounQin 
I will then try to see if I can definitely return an instance of the `type`